lambda/matching.ml

```diff
@@ -1120,19 +1120,6 @@ and split_constr cls args def k =
                       (idef, next) :: nexts )
               )
           )
-      (* QUESTION: split_noex seems to assume that the non-group cases
-         are only variables. How is that correct in presence of GADTs?
-
-         PARTIAL ANSWER: all the split/precompile functions only do work on the
-         first column of a matching.
-         And I think we can assume that, contrary to parmatch, the first
-         column is always well typed.
-         The issue we had in parmatch was due to "default clauses" ending
-         up in subsequent specialized matrices (cf. comment on
-         Parmatch.all_coherent).
-         This doesn't happen here: variables/omegas are in a dedicated group,
-         so rows with omegas at the head end up in a dedicated matrix.
-      *)
```

Comment:
Thinking again about this, I'm not sure that it was the right choice to remove that comment. I propose to keep a less WIP-y version of it. I started writing one, but I think we may want to discuss that at the toplevel instead of in a comment.

Reply:
Ah yes.
I was planning to add an explanation of the whole `split_and_precompile` mechanism, which would make the invariants explicit… but it looks like I forgot.
Feel free to go ahead, let me know when there's something to look at.

Reply:
And now that things are paging back in, what I had in mind was indeed more like a toplevel comment describe the whole flow of the pattern matching compiler, that is:
```
take a "case list"
split and precompile:
  does this and that splitting along the first column, as a result you get
  a list of matrices such that blabla
then there is "proper compilation" of the first column
  divide: does blablabla
  then something something (recursion)
  then combine does this and that
    and in particular for constructors then blabla
done
```